### PR TITLE
Add basic SimCLR model

### PR DESCRIPTION
**Pull request by tnw-chatc**
## Change Description

- [ ] My PR includes a link to the issue that I am addressing

## Solution Description

- Added `SimCLR` model to the Hyrax built-in model registry.
  - This version of `SimCLR` is very basic, but it is functional. The feature extractor is based on a default `resnet18`. The feature extractor can be changed to improve the model (e.g., in a future PR).
  - `NTXentLoss` is used as the default loss function of the model. The implementation is based on [Chen, 2020](https://arxiv.org/abs/2002.05709)

### CODE REVIEW

**Comment by aritraghsh09** [comment]
This looks good to me to merge; but please wait for the review from @mtauraso before merging.

File: src/hyrax/models/simclr.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        super().__init__()
+        self.temperature = temperature
+        self.criterion = nn.CrossEntropyLoss(reduction="sum")
+
+    def forward(self, z_i, z_j):
+        """Forward function of NTXentLoss. Based on Chen, 2020.
+        Loss is calculated from representations from two augmented views of the same batch.
+        """
+        batch_size = z_i.shape[0]
+        device = z_i.device
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/hyrax/models/simclr.py:26]: Oh, ok. Sounds like this is required then! :)

File: src/hyrax/models/simclr.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        self.transform = transform
+
+    def __call__(self, x):
+        x = (x + 1) / 2  # to [0, 1]
+        x = self.transform(x)
+        return x * 2 - 1  # back to (-1, 1)
+
+
+@hyrax_model
+class SimCLRModel(nn.Module):
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/hyrax/models/simclr.py]: Looks good.

**Review by drewoldag — APPROVED** [review]
Thanks for making the updates, I think looks good to go now.

**Comment by aritraghsh09** [comment]
@tnw-chatc I will go ahead and mrege this into `model_updates` for now. If, after running the test, I mentioned in https://github.com/lincc-frameworks/hyrax/issues/381 , you want to rename the loss function; just open another PR.
